cmd/argocd/commands/app.go

```diff
@@ -1351,14 +1352,22 @@ func NewApplicationSyncCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 			errors.CheckError(err)
 
 			appNames := args
-			if selector != "" {
-				list, err := appIf.List(context.Background(), &applicationpkg.ApplicationQuery{Selector: selector})
+			if len(appNames) == 0 {
```

Comment:
It looks as if previously, `-l` was additive. In other words, if I specified `app-name -l some-selector`, I would sync both `app-name` and any apps matching the selector.

With this behavior, `-l` only works if I have not also specified specific app names.

I'd avoid the breaking change, in case this is used in scripts.

Reply:
Makes sense. Thanks for catching it. I've updated the conditional

Reply:
Will include a unit test